### PR TITLE
LPS-178998 | Poshi Automation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# Poshi Automation for Bug LPS-127312 Edit experience modal appears instead of New Experience modal

Hi! :)
This pr is corresponding to this ticket: [LPS-178998](https://issues.liferay.com/browse/LPS-178998)

### CODE REVIEW

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/ExperienceCreateExperience.testcase [hunk 1]
```diff
@@ -5394,4 +5394,38 @@ definition {
 		}
 	}
 
+	@description = "LPS-127312 Bug: Edit experience modal appears instead of New Experience modal"
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/ExperienceCreateExperience.testcase:5397]: The pattern of description we use **This is a test for LPS-127312. The page editor could add new experience after cancel edit experience**

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/ExperienceCreateExperience.testcase [hunk 2]
```diff
@@ -5394,4 +5394,38 @@ definition {
...
+	@priority = 4
+	test ViewExperienceNameNotPersistWithCancelButton {
+		task ("Go to Home page and create a new experience") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/ExperienceCreateExperience.testcase:5400]: The pattern of task blocks we use GWT. **Given a condition, When execute an action, Then assert the results**

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/ExperienceCreateExperience.testcase [hunk 3]
```diff
@@ -5394,4 +5394,38 @@ definition {
...
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/test-site-name/experience-content-page");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/ExperienceCreateExperience.testcase:5401]: This step is already executed inside setup block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/ExperienceCreateExperience.testcase [hunk 4]
```diff
@@ -5394,4 +5394,38 @@ definition {
...
+			Button.clickCancel();
+		}
+
+		task ("Click on Add new experience and assert that the edited Experience name does not persist") {
+			Click(
+				key_text = "New Experience",
+				locator1 = "Button#ANY");
+
+			AssertTextNotEquals(
+				locator1 = "PageEditor#EXPERIENCE_NAME",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/ExperienceCreateExperience.testcase:5426]: Assert the New Experience modal appear

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/ExperienceCreateExperience.testcase [hunk 5]
```diff
@@ -5394,4 +5394,38 @@ definition {
 		}
 	}
 
+	@description = "LPS-127312 Bug: Edit experience modal appears instead of New Experience modal"
+	@priority = 4
+	test ViewExperienceNameNotPersistWithCancelButton {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/ExperienceCreateExperience.testcase:5399]: AddNewExperienceAfterCancelEditExperience

**Review by Tim-Cao — COMMENTED** [review]
Hi @rafaellajordao, would you mind modifying according to comments? Thanks

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
See changes @rafaellajordao
https://github.com/liferay-echo/liferay-portal/pull/11688/commits/cebbc73a07249e35da894623cb9d9091c2366179

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**